packages/core/src/components/button/button.tsx

```diff
@@ -65,10 +66,13 @@ export class Button implements IxButtonComponent {
 
   @Element() hostElement!: HTMLIxButtonElement;
 
-  /**
-   * Temp. workaround until stencil issue is fixed (https://github.com/ionic-team/stencil/issues/2284)
-   */
-  submitButtonElement?: HTMLButtonElement;
+  internals: ElementInternals | null = null;
+
+  constructor() {
+    if (this.hostElement.attachInternals) {
+      this.internals = this.hostElement.attachInternals();
+    }
+  }
```

Comment:
use stencil lib functionality to get `internals`:
https://stenciljs.com/docs/form-associated#creating-a-form-associated-component

```suggestion
  @AttachInternals() internals!: ElementInternals;
```